### PR TITLE
feat: implement `artifact-caching-proxy` in `buildPlugin` (Linux only for now)

### PR DESCRIPTION
**Pull request by lemeurherve**
Excerpt from https://github.com/jenkins-infra/pipeline-library/pull/458
    
Ref: https://github.com/jenkins-infra/helpdesk/issues/2752

### CODE REVIEW

File: vars/buildPlugin.groovy [hunk 1]
```diff
@@ -94,14 +97,69 @@ def call(Map params = [:]) {
...
+                        settingsSecurityFile = settingsFolder + '/settings-security.xml'
+                        sh "mkdir -p ${settingsFolder}"
+                        masterPassword = sh(script: "set +x; mvn --quiet --encrypt-master-password ${randomPassword}", returnStdout: true)
+
+                        masterPassword = masterPassword.replaceAll('[\\n\\r]*', '')
+                        mavenSettingsSecurity = mavenSettingsSecurity.replace('ENCRYPTED-MASTER-PASSWORD', masterPassword)
+                        writeFile file: settingsSecurityFile, text: mavenSettingsSecurity
+
+                        // Generating settings.xml with proxy config and encrypted basic auth password
+                        serverPassword = sh(script: 'mvn --quiet --encrypt-password $ARTIFACT_CACHING_PROXY_PASSWORD', returnStdout: true)
```
Comment (jglick) [inline vars/buildPlugin.groovy:146]: Unsafe.

File: vars/buildPlugin.groovy [hunk 2]
```diff
@@ -94,14 +97,69 @@ def call(Map params = [:]) {
...
                   m2repo = "${pwd tmp: true}/m2repo"
+                  String settingsFile
                   List<String> mavenOptions = [
                     '--update-snapshots',
                     "-Dmaven.repo.local=$m2repo",
                     '-Dmaven.test.failure.ignore',
                     '-Dspotbugs.failOnError=false',
                     '-Dcheckstyle.failOnViolation=false',
                     '-Dcheckstyle.failsOnError=false',
+                    '-X',
```
[the remark this answers is not in the document]
Comment (lemeurherve) [reply vars/buildPlugin.groovy:108]: Yes ofc ^^ Forgot to mention it.

**Comment by lemeurherve** [comment]
Thanks for the feedback @jglick, that what I feared a little bit, I was complicating my life needlessly about encrypting this file.

Rework of this pull request in progress.

**Comment by lemeurherve** [comment]
Superseded by https://github.com/jenkins-infra/pipeline-library/pull/499

**Pull request closed without merging**